### ISSUE TITLE
Feedback: Pinned tabs lose a nice property they had when horizontal

### ISSUE THREAD

**Issue by chutten** [bwinton/VerticalTabs#112]
When horizontal, pinned tabs would remain in the tab bar in a fixed partition while the rest of the tabs were scrolled. In VerticalTabs they don't.

I think it'd be nice to do that here: place pinned tabs in a fixed partition.

Not sure what the corner cases look like (ie, when there are more pinned tabs than width to show unpinned tabs...)... but I think it'd be nice to emulate this behaviour

**Comment by johngruen** [comment]
this would be cool. definitely out of scope for launch, but we're going to be collecting and triaging UI improvements based on bugs and our metrics pipeline. Thanks @chutten

**Comment by hwine** [comment]
@johngruen I actually consider this a regression as "pinned" tabs no longer are visually pinned, they are only "can't accidentally close" pinned. (I suspect most users consider the visual pinning at least as important.)

**Comment by ugggf** [comment]
I vote for the previous behaviour (pinned tabs not scrolled). Or at least an option for it.

**Comment by Zaggy1024** [comment]
It should be possible to fit a huge number of pinned tabs on the sidebar at once if they are placed horizontally next to each other with just the icon (no tab title). The box that contains the pinned tabs could expand downward to fit as many tabs as necessary. I believe this is the behavior in most tree tab extensions, as well.

**Comment by gsurrel** [comment]
The pinned tabs as seen [here](https://mozilla.invisionapp.com/share/GT22ZN6QW#/screens/59200974) are pretty nice!
I'd love that look&feel!

**Comment by chutten** [comment]
> The pinned tabs as seen here are pretty nice!
> I'd love that look&feel!

I like the look, but the change between "open" (where the pinned tabs are horizontal) and "peek" (where the pinned tabs are vertical) will change the location of the tabs and the pinned tabs, decreasing muscle memory (horizontal'll do that to you).

Also, I'm not sure how it would look if the pinned row were to have fewer (say, two) or lots of (say, twenty) pinned tabs.

**Comment by Lurtzae** [comment]
Am I missing an option or is this not implemented yet? This is my number one reason for not using tab center constantly right now. If pinned tabs disappear behind normal tabs it defeats their purpose from my point of view.

**Comment by bwinton** [comment]
@Lurtzae: Yeah, no.  It's not implemented, and not on the roadmap to be implemented any time soon.  I'm sorry that this is a strict requirement for you, but perhaps one of the other vertical tab add-ons will better support your needs.  Thanks for the feedback!  🙂

**Comment by kkot** [comment]
@Lurtzae You can try the option to open new tabs at top. I started using it because of this pinned tabs issue so that I don't need to scroll too much and I really like it - it seams natural after a while.

**Comment by axelsimon** [comment]
> @Lurtzae: Yeah, no. It's not implemented, and not on the roadmap to be implemented any time soon. I'm sorry that this is a strict requirement for you, but perhaps one of the other vertical tab add-ons will better support your needs. Thanks for the feedback! 🙂
Is this still the case? One could consider TabCenter breaks built-in Firefox functionality, which sounds like a big no-no to me.

Otherwise +1 with @Lurtzae. And great extension otherwise :)
